internal/controllers/utils/provision.go

```diff
@@ -262,6 +265,127 @@ func ValidateDefaultInterfaces[T any](data T) error {
 	return nil
 }
 
+// removeLabelFromInterfaces removed the label property for each interface as the label
+// property is not part of the ClusterInstance schema.
+func removeLabelFromInterfaces[T any](data T) error {
+	dataMap, _ := any(data).(map[string]any)
+	nodes, ok := dataMap["nodes"].([]any)
+	if ok {
+		for _, node := range nodes {
+			nodeMap, ok := node.(map[string]interface{})
+			if !ok {
+				return fmt.Errorf("unexpected: invalid node data structure")
+			}
+			interfaces := getInterfaces(nodeMap)
+			if interfaces == nil {
+				return fmt.Errorf("failed to extract the interfaces from the node map")
+			}
+			for _, intf := range interfaces {
+				if value, exists := intf["label"]; exists && value != "" {
```

Comment:
should the label be removed if its value is empty? It seems like this `if` check can be removed since if it is nil or there is no such element, `delete` is a no-op.

Reply:
Right, we can remove the check and do the delete anyway, it won't be an overhead.